src/wordy/ast/BlockNode.java

```diff
@@ -60,4 +65,23 @@ protected String describeAttributes() {
         return "(%d %s)"
             .formatted(statements.size(), statements.size() == 1 ? "child" : "children");
     }
+
+    @Override
+    protected void doRun(EvaluationContext context) {
+        for(var statement : statements) {
+            statement.run(context);
+        }
+    }
+
+    @Override
+    public void compile(PrintWriter out) {
+        out.print("{ ");
+        for (int i = 0; i < statements.size(); i++) {
+            statements.get(i).compile(out);
+            if (i < statements.size() - 1) {
```

Comment:
same here, trying to cover an edge case, but looking at your solution, I think I can simplify it.